### PR TITLE
Initial resource class work

### PR DESCRIPTION
**Pull request by mtaylor**
Initial work to for resource class creation.  

N.B. We still need to add in relationships between resource class and flavors and racks.  This will follow in subsequent pull request.

N.B. Be sure to remove existing sqlite db and run: 

```
$ tuskar-dbsync --config-file etc/tuskar/tuskar.conf --debug
```

To test manually, run server:

```
$ tuskar-api --config-file etc/tuskar/tuskar.conf
```

To create resource class:

```
$ curl -H "accept: application/json" -H "content-type: application/json" -d '{"name":"foobar", "service_type":"compute"}' http://0.0.0.0:6385/v1/resource_classes
```

To list all resource classes:

```
$ curl -H "accept: application/json" http://0.0.0.0:6385/v1/resource_classes
```

### CODE REVIEW

**Comment by mfojtik** [comment]
Looks good to me. ACK.

File: tuskar/db/sqlalchemy/api.py [hunk 1]
```diff
@@ -93,6 +93,18 @@ def create_rack(self, values):
         rack.save()
         return rack
 
+    def get_resource_classes(self, columns):
+        session = get_session()
+        return session.query(models.ResourceClass).all()
+
+    def create_resource_class(self, values):
+        rc = models.ResourceClass()
+        # FIXME: This should be DB transaction ;-)
```
Comment (markmc) [inline tuskar/db/sqlalchemy/api.py:102]: What else should be in the transaction? Don't follow the comment

Also, format for FIXMEs is # FIXME(mtaylor): ...

File: tuskar/api/controllers/v1.py [hunk 2]
```diff
@@ -165,3 +192,5 @@ class Controller(object):
     blaas = BlaasController()
 
     racks = RacksController()
+
+    resource_classes = ResourceClassesController()
```
Comment (markmc) [inline tuskar/api/controllers/v1.py:196]: Missing newline at EOF

File: tuskar/db/sqlalchemy/models.py [hunk 3]
```diff
@@ -143,3 +143,15 @@ def convert_capacity(c):
...
+    """Represents a Resource Class."""
+
+    __tablename__ = 'resource_classes'
+    id = Column(Integer, primary_key=True)
+    name = Column(Text, unique=True)
+    service_type = Column(Text, unique=True)
+
+    def as_dict(self):
+        d = super(ResourceClass, self).as_dict()
+        return d
```
Comment (markmc) [inline tuskar/db/sqlalchemy/models.py:157]: Missing newline at EOF

**Comment by markmc** [comment]
Nice work

**Pull request merged**